features/mbedtls/platform/src/platform_alt.c

```diff
@@ -20,38 +20,33 @@
 
 #include "mbedtls/platform.h"
 #if defined(MBEDTLS_PLATFORM_SETUP_TEARDOWN_ALT)
-#include "sns_silib.h"
+mbedtls_platform_context ctx = {0};
 
-/* once https://github.com/ARMmbed/mbedtls/issues/1200 will be supported,
- * rndState should be part of mbedtls_platform_context
- * Until then, we should keep it global and extern */
-
-CRYS_RND_State_t     rndState = { { 0 } } ;
-CRYS_RND_WorkBuff_t  rndWorkBuff = { { 0 } } ;
-
-
-int mbedtls_platform_setup( mbedtls_platform_context *ctx )
+int mbedtls_platform_setup( mbedtls_platform_context *obsolete_ctx )
 {
     int ret = 0;
-    if( ctx == NULL )
-        return ( -1 );
 
-    /* call platform specific code to setup CC driver*/
-    if( ( ret = cc_platform_setup( &ctx->platform_impl_ctx ) ) != 0 )
-        return ( ret );
+    ctx.reference_count++;
```

Comment:
Does this function need to work in multi-threaded environments (multiple callers from different threads)?

If so, consider using a mutex or atomic increment (`core_util_atomic_incr_u32()`). Otherwise, we risk calling setup multiple times if two threads overlap calls to `mbedtls_platform_setup()`. Both threads can read reference count of 0 into a register, increment the register copy, and write back the register to memory at the same time. Then both will proceed to call `crypto_platform_setup()`.

Reply:
I did leave an open question in the Pr description, whether we should consider concurrency issues;
>Question: Should we protect the reference counter with a mutex?

I'll change to atomic increase and decrease